pandas/core/base.py

```diff
@@ -1270,6 +1271,13 @@ def searchsorted(
         sorter: NumpySorter = None,
     ) -> npt.NDArray[np.intp] | np.intp:
 
+        if isinstance(value, pd.DataFrame):
```

Comment:
check with ABCDataFrame

Reply:
I am not sure about the usage of ABCDataFrame. Does that mean I should use `isinstance(value, ABCDataFrame)` instead?

Reply:
yes.  at the top do `from pandas.core.dtypes.generic import ABCDataFrame`